### PR TITLE
For release of VF

### PR DESCRIPTION
**Pull request by fosterlynn**
first round of json-schemas, please review @bhaugen

### CODE REVIEW

File: schemas/Action.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/Action.json",
+    "title": "Action",
+    "description": "An action verb defining the kind of event, commitment, or intent.",
```
Comment (bhaugen) [inline schemas/Action.json:5]: "Defining" seems more total while action is a particularity of events etc. Maybe
"defining what an event, commitment, or intent, does or commits or intends to do"?

Below, you wrote 
"Relates a process input or output to a verb, such as consume, produce, work, improve, etc."
So could be
"Relates an event, commitment, or intent, to a verb, such as consume, produce, work, improve, etc.",

File: schemas/Action.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            "type": "string",
+            "enum": ["dropoff", "pickup", "consume", "use", "work", "cite", "produce", "accept", "modify", "pass", "fail", "deliver-service", "transfer-all-rights", "transfer-custody", "transfer-complete", "move", "raise", "lower"]
+        },
+        "resourceEffect": {
+            "description": "The effect of an economic event on a resource, increment, decrement, no effect, or decrement resource and increment 'to' resource.",
+            "type": "string",
+            "enum": ["+", "-", "0", "-+"]
+        }
+    },
+    "required": ["name"]
```
Comment (bhaugen) [inline schemas/Action.json]: resourceEffect is not required? What if it does not exist? Should it have a default?

File: schemas/Action.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/Action.json",
+    "title": "Action",
+    "description": "An action verb defining the kind of event, commitment, or intent.",
+    "type": "object",
+    "properties": {
+        "name": {
+            "description": "An informal or formal textual identifier for an agent. Does not imply uniqueness.",
```
Comment (bhaugen) [inline schemas/Action.json]: oops, agent s/b action.

File: schemas/AgentRelationshipRole.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/AgentRelationshipRole.json",
+    "title": "AgentRelationshipRole",
+    "description": "A relationship role defining the kind of association one agent can have with another.",
+    "type": "object",
+    "properties": {
+        "label": {
+            "description": "The human readable name of the role, from the subject to the object.",
```
Comment (bhaugen) [inline schemas/AgentRelationshipRole.json]: Add example. Also to inverseLabel.

File: schemas/Agreement.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/Agreement.json",
+    "title": "Agreement",
+    "description": "Any type of agreement among economic agents.",
+    "type": "object",
+    "properties": {
+        "name": {
+            "description": "An informal or formal textual identifier for the item. Does not imply uniqueness.",
```
Comment (bhaugen) [inline schemas/Agreement.json:9]: In most cases, you are repeating the schema in the name description. In this case, just "for the item", not "for the Agreement".

File: schemas/Claim.json [hunk 6]
```diff
@@ -0,0 +1,72 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/Claim.json",
+    "title": "Claim",
+    "description": "A claim for a future economic event(s) in reciprocity for an economic event that already occurred. For example, a claim for payment for goods received.",
+    "type": "object",
+    "properties": {
+        "action": {
+            "description": "Relates a process input or output to a verb, such as consume, produce, work, improve, etc.",
```
Comment (bhaugen) [inline schemas/Claim.json]: Relates a process input or output to a verb?
S/b 
"Relates a claim to a verb"?

But should a claim even have an action? If so, why?

File: schemas/Commitment.json [hunk 7]
```diff
@@ -0,0 +1,115 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/Commitment.json",
+    "title": "Commitment",
+    "description": "A planned economic flow that has been promised by an agent to another agent.",
+    "type": "object",
+    "properties": {
+        "action": {
+            "description": "Relates a process input or output to a verb, such as consume, produce, work, improve, etc.",
```
Comment (bhaugen) [inline schemas/Commitment.json]: "Relates a process input or output to a verb
s/b 
"Relates a commitment to a verb?

File: schemas/Intent.json [hunk 8]
```diff
@@ -0,0 +1,103 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/Intent.json",
+    "title": "Intent",
+    "description": "A planned economic flow, which can lead to economic events (sometimes through commitments).",
+    "type": "object",
+    "properties": {
+        "action": {
+            "description": "Relates a process input or output to a verb, such as consume, produce, work, improve, etc.",
```
Comment (bhaugen) [inline schemas/Intent.json]: "Relates a process input or output to a verb
s/b
"Relates an intent to a verb ?

File: schemas/Claim.json [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        },
+        "resourceQuantity": {
+            "description": "The amount and unit of the economic resource counted or inventoried.",
+            "$ref": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/QuantityValue.json"
+        },
+        "effortQuantity": {
+            "description": "The amount and unit of the work or use or citation effort-based action. This is often a time duration, but also could be cycle counts or other measures of effort or usefulness.",
+            "$ref": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/QuantityValue.json"
+        },
+        "resourceClassifiedAs": {
```
Comment (bhaugen) [inline schemas/Claim.json:20]: Why does a claim have a resource? I looked at NRP, our claims do not have resources.

File: schemas/Claim.json [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+            "description": "References a concept in a common taxonomy or other classification scheme for purposes of categorization or grouping.",
+            "type": "array",
+                "items": {
+                    "type": "string",
+                    "format": "uri"
+                }
+        },
+        "resourceConformsTo": {
```
Comment (bhaugen) [inline schemas/Claim.json:28]: Same comment as above.

File: schemas/EconomicEvent.json [hunk 11]
```diff
@@ -1,71 +1,107 @@
...
     "properties": {
         "action": {
             "description": "Relates a process input or output to a verb, such as consume, produce, work, improve, etc.",
             "type": "string"
         },
-        "quantifiedAs": {
-            "description": "The quantity and unit of the event. This is the quantity that could be used to increment or decrement a resource, depending on the type of resource and type of event.",
-            "$ref": "https://raw.githubusercontent.com/valueflows/vf-json-schema/master/schemas/QuantityValue.json"
+        "resourceQuantity": {
+            "description": "The amount and unit of the economic resource counted or inventoried.",
```
Comment (bhaugen) [inline schemas/EconomicEvent.json]: In the previous description, you added (and I reword slightly here): 
This is the quantity that could be used to increment or decrement a resource, depending on the type of resource and resource effect of action.",

File: schemas/EconomicEvent.json [hunk 12]
```diff
@@ -1,71 +1,107 @@
...
             "type": "string",
             "format": "date-time"
         },
         "hasEnd": {
-            "description": "The ending date and optionally time of the economic event.",
+            "description": "Specific time marking the exact end of the economic event.",
             "type": "string",
             "format": "date-time"
         },
         "hasPointInTime": {
```
Comment (bhaugen) [inline schemas/EconomicEvent.json:62]: Is hasPointInTime an alternative to hasBeginning and hasEnd? If so, wd  be good to say so.

File: schemas/EconomicEvent.json [hunk 13]
```diff
@@ -1,71 +1,107 @@
...
-            "description": "The economic event occurs at this date/time instant.",
+            "description": "Specific time marking the exact time at which the economic event occurred.",
             "type": "string",
             "format": "date-time"
         },
         "before": {
```
Comment (bhaugen) [inline schemas/EconomicEvent.json:67]: What is the use of before and after in relation to beginning, end, and pointInTime?

File: schemas/EconomicEvent.json [hunk 14]
```diff
@@ -1,71 +1,107 @@
...
+            "$ref": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/SpatialThing.json"
+        },
         "note": {
             "description": "A textual comment or description.",
             "type": "string"
         },
         "inputOf": {
-            "description": "The process for which this economic event describes an input.",
-            "$ref": "https://raw.githubusercontent.com/valueflows/vf-json-schema/master/schemas/Process.json"
+            "description": "Defines the process to which this event is an input.",
```
Comment (bhaugen) [inline schemas/EconomicEvent.json:86]: Sometimes you say "Defines" and sometimes "References" for a $ref.

File: schemas/EconomicEvent.json [hunk 15]
```diff
@@ -1,71 +1,107 @@
...
+        },
+        "realizationOf": {
+            "description": "This economic event occurs as part of this agreement.",
+            "$ref": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/Agreement.json"
         },
-        "under": {
-            "description": "Reference to an agreement between agents which specifies the rules or policies which govern this event.",
-            "$ref": "https://raw.githubusercontent.com/valueflows/vf-json-schema/master/schemas/Agreement.json"
+        "agreedIn": {
+            "description": "Reference to an agreement between agents which specifies the rules or policies or calculations which govern this economic event.",
```
Comment (bhaugen) [inline schemas/EconomicEvent.json:102]: The word "calculation" is probly used in other agreement references, but this is the first time it triggered the question of what does that mean?

File: schemas/Process.json [hunk 16]
```diff
@@ -1,66 +1,70 @@
...
-            },
-            "minItems": 1,
-            "uniqueItems": true
+        "basedOn": {
+            "description": "The recipe definition or specification for a process.",
+            "$ref": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/ProcessSpecification.json"
         },
-        "committedInputs": {
-            "description": "Planned commitments that bring tangible or intangible resources into the process.",
+        "classifiedAs": {
```
Comment (bhaugen) [inline schemas/Process.json]: If I understand correctly, I see 2 classifiedAs in this schema....?

File: schemas/RecipeFlow.json [hunk 17]
```diff
@@ -0,0 +1,50 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/RecipeFlow.json",
+    "title": "RecipeFlow",
+    "description": "The linkage between a recipe process, an action that structures a recipe, and a resource specification.",
```
Comment (bhaugen) [inline schemas/RecipeFlow.json]: This description was hard to parse. And too geeky.
How about:
The specification of a resource inflow to, or outflow from, a recipe process.

File: schemas/RecipeFlow.json [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+        },
+        "effortQuantity": {
+            "description": "The amount and unit of the work or use or citation effort-based action. This is often a time duration, but also could be cycle counts or other measures of effort or usefulness.",
+            "$ref": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/QuantityValue.json"
+        },
+        "recipeFlowResource": {
+            "description": "The resource definition referenced by this flow in the recipe.",
+            "$ref": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/RecipeResource.json"
+        },
+        "recipeInputOf": {
```
Comment (bhaugen) [inline schemas/RecipeFlow.json:28]: This and the next attribute name are not self-defining. They mean input output of a recipeProcess, but only the description says so.

File: schemas/Commitment.json [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+                    "type": "string",
+                    "format": "uri"
+                }
+        },
+        "resourceConformsTo": {
+            "description": "The primary resource specification or definition of an existing or potential economic resource. A resource will have only one, as this specifies exactly what the resource is.",
+            "$ref": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/ResourceSpecification.json"
+        },
+        "stage": {
+            "description": "The last stage the desired economic resource went through.",
```
Comment (bhaugen) [inline schemas/Commitment.json]: That should say something like
"References the ProcessSpecification of the last process the desired economic resource went through. Stage is used only for resources that pass as both input and output through more than one process, for example, proofreading, editing, and publishing a document, where the publishing process wants only documents that have gone through the editing process",
in all instances of "last stage".

File: schemas/RecipeProcess.json [hunk 20]
```diff
@@ -0,0 +1,34 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/RecipeProcess.json",
+    "title": "RecipeProcess",
+    "description": "Specifies the process part of a recipe for use in planning from recipe.",
```
Comment (bhaugen) [inline schemas/RecipeProcess.json]: How about
"Specifies a process in a recipe for use in planning from recipe.",

Comment (bhaugen) [reply schemas/RecipeProcess.json]: > How about
> "Specifies a process in a recipe for use in planning from recipe."

As of now, it talks about a "process part". And yes I know you meant "part of a recipe", but this reads like a recipe has something called a process part. Like as opposed to the resource part.

File: schemas/Action.json (hunk 2 above)
Comment (fosterlynn) [reply schemas/Action.json]: I don't think it should have a default, that would just be wrong a lot.  I think some people may just hard code it in logic.  But we could require it too, I'm fine either way.

File: schemas/Scenario.json [hunk 21]
```diff
@@ -0,0 +1,36 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/valueflows/valueflows/master/json-schemas/Scenario.json",
+    "title": "Scenario",
+    "description": "Supports including intents in multiple proposals, as well as a proposal including multiple intents.",
```
Comment (bhaugen) [inline schemas/Scenario.json]: Copy pasta error from ProposedIntent?

File: schemas/Claim.json (hunk 9 above)
Comment (fosterlynn) [reply schemas/Claim.json:20]: not a resource, just classification or specification, should have that, right?  if you are claiming a type of resource

File: schemas/EconomicEvent.json (hunk 13 above)
Comment (fosterlynn) [reply schemas/EconomicEvent.json:67]: usually alternatives, but also might want a due date plus a planned interval, say - just gives different alternatives for exactness

File: schemas/EconomicEvent.json (hunk 15 above)
Comment (fosterlynn) [reply schemas/EconomicEvent.json:102]: refers to for example values equations / distribution agreements - the calculation explains why you see that in a claim

File: schemas/EconomicEvent.json (hunk 13 above)
Comment (bhaugen) [reply schemas/EconomicEvent.json:67]: Possible to explain succinctly somewhere? Otherwise what shd implementers do?
